packages/grid/_modules_/grid/hooks/features/export/seralizers/csvSeraliser.ts

```diff
@@ -15,37 +14,38 @@ const serialiseCellValue = (value) => {
 };
 
 export function serialiseRow(
-  row: GridRowModel,
+  id: GridRowId,
   columns: GridColumns,
   getCellValue: (id: GridRowId, field: string) => GridCellValue,
 ): Array<string> {
   const mappedRow: string[] = [];
   columns.forEach(
     (column) =>
       column.field !== gridCheckboxSelectionColDef.field &&
-      mappedRow.push(serialiseCellValue(getCellValue(row.id, column.field))),
+      mappedRow.push(serialiseCellValue(getCellValue(id, column.field))),
   );
   return mappedRow;
 }
 
 export function buildCSV(
   columns: GridColumns,
-  rows: GridRowModel[],
+  rowIds: GridRowId[],
   selectedRows: Record<GridRowId, boolean>,
```

Comment:
not sure I understand why this one doesn't just get a set of rows. It should be dumb and selection agnostic.